.changeset/three-emus-tell.md

```diff
@@ -0,0 +1,116 @@
+- `DatePickerRangePanel` props `StartCalendarWeekHeaderProps` and `EndCalendarWeekHeaderProps` removed.
+  Pass props to `CalendarWeekHeader` using `StartCalendarGridProps -> CalendarWeekHeaderProps` and `EndCalendarGridProps -> CalendarWeekHeaderProps`.
+
+```
+<DatePicker
+  selectionVariant="range"
+>
+  <DatePickerTrigger>
+    <DatePickerRangeInput />
+  </DatePickerTrigger>
+  <DatePickerOverlay>
+    <DatePickerRangePanel
+     StartCalendarGridProps={{ CalendarWeekHeaderProps: { "data-test-id" : "some value" }}}
+     EndCalendarGridProps={{ CalendarWeekHeaderProps: { "data-test-id" : "some value" }}}
+     />
+  </DatePickerOverlay>
+</DatePicker>
+```
+
+- `DatePicker` now supports `isDayDisabled`, `isDayUnselectable`, and `isDayHighlighted` props, rather than providing them through the `Calendar` props.
+- The `DatePickerOverlayProvider` prop `onOpenChange` has changed signature and `event` has been removed, to create a consistent API with other components.
+
+```diff
+- onOpenChange?: ( newOpen: boolean, event?: Event, reason?: DatePickerOpenChangeReason) => void;
++ onOpenChange?: ( newOpen: boolean, reason?: DatePickerOpenChangeReason) => void;
+```
+
+- DatePicker helper text simplified and resizing bug fixed
+
+In our DatePicker examples, we demonstrate how to include helper text when using a DatePicker within a FormField.
+The design requires the helper text to move inside the panel when the DatePicker is opened.
+However, this caused a resizing issue with the original date input.
+
+To resolve this, we have refactored the implementation by introducing a `DatePickerHelperText` component.
+This component is used as a child of the `DatePicker`, streamlining the process and removing the need for users to manage the visibility of the helper text themselves.
+
+```diff
+- <FormField validationStatus={validationStatus}>
+-   <FormLabel>Select a date</FormLabel>
+-   <DatePicker
+-     selectionVariant={"single"}
+-   >
+-     <DatePickerTrigger>
+-       <DatePickerSingleInput />
+-     </DatePickerTrigger>
+-     <DatePickerOverlay>
+-       <DatePickerSingleGridPanel
+-         helperText={helperText}
+-       />
+-       <DatePickerHelperText>{helperText}</DatePickerHelperText>
+-     </DatePickerOverlay>
+-   </DatePicker>
+-   {!open ? <FormHelperText>{helperText}</FormHelperText> : null}
+- </FormField>
++ <FormField validationStatus={validationStatus}>
++   <FormLabel>Select a date</FormLabel>
++   <DatePicker
++     selectionVariant={"single"}
++   >
++     <DatePickerTrigger>
++       <DatePickerSingleInput />
++     </DatePickerTrigger>
++     <DatePickerOverlay>
++       <DatePickerSingleGridPanel
++         helperText={helperText}
++       />
++     </DatePickerOverlay>
++     <DatePickerHelperText>{helperText}</DatePickerHelperText>
++   </DatePicker>
++ </FormField>
+```
+
+- Added `useFocusOut` floating-ui middleware to close the overlay upon focus out.
```

Comment:
This may introduces some problem. On the site, go to first single date example. 
Focus in input > Tab (focus on calendar icon) > Space (open overlay) > Tab (overlay closed), so no way to move focus to the overlay via keyboard?

Reply:
this was fixed with the focus changes, opening the overlay with space, moves focus to the overlay and then your in a focus trap, escape to close etc.